js/components/MilkdropWindow/index.js

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import butterchurn from "butterchurn";
+import butterchurnPresets from "butterchurn-presets";
+import GenWindow from "../GenWindow";
+
+class MilkdropWindow extends React.Component {
+  constructor() {
+    super();
+  }
+  componentDidMount() {
+    const analyserNode = this.props.mediaPlayer._analyser;
+    const gainNode = this.props.mediaPlayer._gainNode;
+    this.visualizer = butterchurn.createVisualizer(
+      analyserNode.context,
+      this._canvasNode,
+      {
+        width: this._canvasNode.width,
+        height: this._canvasNode.height
+      }
+    );
+    this.visualizer.connectAudio(gainNode);
```

Comment:
Do you just use this to connect an analyser node? If so, would it make sense to expose a method that accepted the analyser itself, since we already have it?

Reply:
Yea, I am just using it to create an analyser.  It seems like our analysers are slightly different (FFT size for example), so it might be easier to keep them separate, although I guess the overhead of multiple audio analyses would be nice to avoid.

Reply:
Ahh, good point about the analyser's config. I'll have to get MilkDrop working locally to test this, but if I recall correctly, visualizers were always either or. The default little ones wold disable if you had a "real" visualizer going. I wonder what the overhead of an analyser actually is if it's never queried.

Maybe the best bet is to just expose the context (to create a new analyser) and a "connect" method. Some smarts to ensure only one visualizer is enabled at the same time might be nice, but I'm not sure how off the top of my head. I guess you would have to register your window as a visualizer.

I guess it won't be long before we're reimplementing all of Winamp's plugin API.